app/mailers/delayed_mailer.rb

```diff
@@ -13,113 +14,101 @@ def initialize(assignment_id, deadline_type, due_at)
+      end
+
+      if self.deadline_type == "drop_outstanding_reviews"
+        drop_outstanding_reviews
       end
     end
   end
 
-  # Find the signed_up_teams of the assignment and send mails to them
   def mail_signed_up_users
-    emails = []
-    assignment = Assignment.find(self.assignment_id)
     sign_up_topics = SignUpTopic.where(['assignment_id = ?', self.assignment_id])
+    emails = if sign_up_topics.nil? || sign_up_topics.count.zero?
+               find_team_members_email
+             else
+               find_team_members_email_for_all_topics(sign_up_topics)
+             end
+    email_reminder(emails, self.deadline_type) if emails and !emails.empty?
+  end
 
-    # If there are sign_up topics for an assignement then send a mail toonly signed_up_teams else send a mail to all participants
-    if (sign_up_topics.nil? || sign_up_topics.count == 0)
-      if assignment.team_assignment?
-        teamMails = getTeamMembersMail
-        for mail in teamMails
-          emails << mail
-          email_reminder(emails, self.deadline_type)
-        end
-      else
-        mail_assignment_participants
-      end
-    else
-      for topic in sign_up_topics
-        signedUpTeams = SignedUpTeam.where(['topic_id = ?', topic.id])
-        unless assignment.team_assignment?
-          for signedUser in signedUpTeams
-            uid  = signedUser.team_id
-            user = User.find(uid)
-            emails << user.email
-          end
-        else
-          for signedUser in signedUpTeams
-            teamid = signedUser.team_id
-            team_members = TeamsUser.where(team_id: teamid)
-            for team_member in team_members
-              user = User.find(team_member.user_id)
-              emails << user.email
-            end
-          end
-        end
+  def find_team_members_email
+    emails = []
+    teams = Team.where(['parent_id = ?', self.assignment_id])
+    for team in teams
+      for team_member in team.users
+        emails << team_member.email
       end
-      email_reminder(emails, self.deadline_type)
-    end
     end
+    emails
+  end
 
-  def getTeamMembersMail
-    teamMembersMailList = []
-    assignment = Assignment.find(self.assignment_id)
-    teams = Team.where(['parent_id = ?', self.assignment_id])
-    for team in teams
-      team_participants = TeamsUser.where(['team_id = ?', team.id])
-      for team_participant in team_participants
-        user = User.find(team_participant.user_id)
-        teamMembersMailList << user.email
+  def find_team_members_email_for_all_topics(sign_up_topics)
+    emails = []
+    sign_up_topics = [sign_up_topics] unless sign_up_topics.respond_to?(:each)
+    for sign_up_topic in sign_up_topics
+      for signed_up_team in sign_up_topic.signed_up_teams
+        for user in signed_up_team.team.users
+          emails << user.email
+        end
       end
     end
-    teamMembersMailList
+    emails
   end
 
   def get_one_member_team
```

Comment:
This is a bad name. It gets the emails of one-person-teams who hold topics. In addition, method name starting with "get" is not suggested for ruby.